Update SDK

MackinnonBuck opened the pull request.

MackinnonBuck committed:
Update SDK version

MackinnonBuck committed:
Update SDK version

MackinnonBuck commented:
The `aspnetcore-components-e2e` and `aspnetcore-ci (Build Source-Build (Managed))` checks have failed twice now because `dotnet-install` failed:
```
/home/vsts/.nuget/packages/microsoft.dotnet.arcade.sdk/7.0.0-beta.22379.10/tools/InstallDotNetCore.targets(15,5): error : dotnet-install failed [/home/vsts/.nuget/packages/microsoft.dotnet.arcade.sdk/7.0.0-beta.22379.10/tools/Tools.proj]
```
Not totally sure what the cause is at the moment. See https://dev.azure.com/dnceng/public/_build/results?buildId=1916161&view=logs&j=2f0d093c-1064-5c86-fc5b-b7b1eca8e66a&t=d2b639a6-cb19-5f1f-66fd-8047f66b3026&l=126

wtgodbe commented:
I'd recommend checking out the binlog to see if there's more info ([here](https://artprodcus3.artifacts.visualstudio.com/Ab55de4ed-4b5a-4215-a8e4-0a0a5f71e7d8/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2RuY2VuZy9wcm9qZWN0SWQvOWVlNmQ0NzgtZDI4OC00N2Y3LWFhY2MtZjZlNmQwODJhZTZkL2J1aWxkSWQvMTkxNjE2MS9hcnRpZmFjdE5hbWUvQnVpbGRMb2dzX1NvdXJjZUJ1aWxkX01hbmFnZWRfQXR0ZW1wdDE1/content?format=file&subPath=%2Fartifacts%2Flog%2FRelease%2FBuild.binlog)). Looks like it's only in the source-build leg, so maybe there was a relevant change there in dotnet/sdk or dotnet/installer recently. @MichaelSimons might know more

MichaelSimons commented:
cc @dotnet/source-build-internal

MackinnonBuck committed:
Update SDK to 7.0.100-rc.1.22402.2

dougbu commented:
/azp run

SteveSandersonMS committed:
Updating further

Tratcher commented:
https://github.com/dotnet/arcade/issues/10311#issuecomment-1209629471

> As it's now clear this is an MSBuild bug in this SDK (and thus not one users want to adopt) I am closing this issue.

I'll try a later SDK until this works.

Tratcher committed:
Update SDK to 7.0.100-rc.1.22408.5

dougbu commented:
@Tratcher this SDK didn't work. Suggest watching <https://maestro-prod.westus2.cloudapp.azure.com/2237/https:%2F%2Fgithub.com%2Fdotnet%2Finstaller/latest/graph> for a build that includes dotnet/msbuild@f2d13c6261c25bfdf53b29f3b0c4f38d7274fcc4. Latest SDK (which, confusingly, we get from the dotnet/installer repo) includes msbuild commits from 6 days ago but we want a commit from 4 days ago.

Tratcher committed:
Update SDK to 7.0.100-rc.1.22410.4

TanayParikh committed:
7.0.100-rc.1.22404.18

TanayParikh commented:
Updated to `7.0.100-rc.1.22404.18` to match latest SDK from installer.

TanayParikh committed:
7.0.100-rc.1.22410.15

TanayParikh committed:
Update global.json

TanayParikh commented:
> Suggest watching https://maestro-prod.westus2.cloudapp.azure.com/2237/https:%2F%2Fgithub.com%2Fdotnet%2Finstaller/latest/graph for a build that includes [dotnet/msbuild@f2d13c6](https://github.com/dotnet/msbuild/commit/f2d13c6261c25bfdf53b29f3b0c4f38d7274fcc4). Latest SDK (which, confusingly, we get from the dotnet/installer repo) includes msbuild commits from 6 days ago but we want a commit from 4 days ago.

Just did a trace and I believe `7.0.100-rc.1.22411.3` should now have everything we need. 🤞

TanayParikh commented:
> should now have everything we need

😞 unfortunately not. Source build still failing.

TanayParikh committed:
Update global.json

TanayParikh commented:
Looks like we have a new breaking change now:

> src/DataProtection/DataProtection/src/AuthenticatedEncryption/ManagedAuthenticatedEncryptorFactory.cs(116,61): error IL2070: (NETCORE_ENGINEERING_TELEMETRY=Build) 'this' argument does not satisfy 'DynamicallyAccessedMemberTypes.PublicParameterlessConstructor' in call to 'System.Type.MakeGenericType(params Type[])'. The parameter 'implementation' of method 'Microsoft.AspNetCore.DataProtection.AuthenticatedEncryption.ManagedAuthenticatedEncryptorFactory.AlgorithmActivator.CreateFactory<T>(Type)' does not have matching annotations. The source value must declare at least the same requirements as those declared on the target location it is assigned to.

MichaelSimons commented:
> Looks like we have a new breaking change now:

I just discovered this in source-build and logged an issue for it - https://github.com/dotnet/aspnetcore/issues/43253

JamesNK committed:
Validate DataProtection custom algorithm has a constructor

JamesNK commented:
I've added a possible fix for the warning - https://github.com/dotnet/aspnetcore/pull/43259

It might not fix the warning - ASP.NET Core isn't using the same SDK as source build so I can't test it - but if it's still a problem then it should be a minor fix.

dougbu commented:
Might be easier to test your commit in this PR's branch @JamesNK

JamesNK committed:
Additional fixes

JamesNK commented:
@vitek-karas What's going on with IL2121? I tried to remove some of the suppressions, but they were correctly suppressing warnings.

I see you disabled this warning in [dotnet/runtime@`e65381b` (#73410)](https://github.com/dotnet/runtime/pull/73410/commits/e65381b30c5a99edfc4510ab0c029c6222c40d99). Is IL2121 broken? Should it be suppressed in aspnetcore?

JamesNK committed:
Suppress IL2121

vitek-karas commented:
IL2121 is off by default, but that "off by default" part is in SDK - which runtime or asp.net don't use to run the linker 😢 
Please suppress them for now.

That said I would be interested to see the cases where the suppression is necessary but linker produces IL2121 still. The only place where we've seen that happening was either around feature switches, or in code which uses ifdefs.

sebastienros committed:
Update global.json

sebastienros committed:
Update global.json

sebastienros committed:
Update global.json

sebastienros committed:
Fix NoWarn overrides

sebastienros commented:
@vitek-karas there is the suppression but we are still seeing errors. I did check that no suppression is overridden. Example: https://github.com/dotnet/aspnetcore/pull/43028/checks?check_run_id=7864889493

javiercn commented:
@vitek-karas any update here?

javiercn commented:
@sbomer there seems to be an ILinker issue that is triggering warnings despite suppresions, could you take a look at it as per https://github.com/dotnet/aspnetcore/pull/43028#issuecomment-1217090062?

sebastienros committed:
Update LinkabilityChecker.csproj

sebastienros committed:
Update WasmLinkerTest.csproj

sebastienros commented:
@sbomer thanks, I updated these files with `NoWarn="$(NoWarn)"`, let's see if that helps.

tlakollo commented:
I can see that there are no linker warnings being shown in the output, but the installer script is failing with a no-so-useful message

sebastienros commented:
The ILLINK warnings are gone, thanks

sebastienros commented:
@MichaelSimons I see you created https://github.com/dotnet/aspnetcore/issues/43253 for source build issues, do you know if there is something we can do in this PR to get it building? Wait/update to a new sdk?

MichaelSimons commented:
@sebastienros, Regarding https://github.com/dotnet/aspnetcore/issues/43253, that was specific to the linker issue that I see was already addressed in this PR.

In regards to the failing source-build CI leg, the symptoms appear to be similar to the original issue reported in https://github.com/dotnet/arcade/issues/10311.  Perhaps @MattGal can comment on the situation.

vitek-karas commented:
@sebastienros regarding the reported unnecessary suppression - it's complaining about this suppression: https://github.com/dotnet/aspnetcore/blob/61e1572ad518c82f6fd487e65852e41bccde1bc1/src/Components/Components/src/Microsoft.AspNetCore.Components.WarningSuppressions.xml#L64

We've improved the linker to "blame" the XML file in a case like this, but the change probably didn't make it into this repo yet: https://github.com/dotnet/linker/commit/9332578bfd67d8b1e53a3cbf9e0f92d5af7186ac

dougbu commented:
@vitek-karas should I remove the suppression in the meantime❔

vitek-karas commented:
@dougbu if you want to, but with suppression of IL2121 there's nothing to do. Eventually we should go over all of the IL2121 warnings and clean it up. Basically aspnetcore equivalent of https://github.com/dotnet/runtime/pull/73238 in runtime.

The linker change will only produce the warning pointing to the XML file, but it will still produce the warning (just makes it easier to find).

/cc @jkurdek

MattGal commented:
> @sebastienros, Regarding #43253, that was specific to the linker issue that I see was already addressed in this PR.
> 
> In regards to the failing source-build CI leg, the symptoms appear to be similar to the original issue reported in [dotnet/arcade#10311](https://github.com/dotnet/arcade/issues/10311). Perhaps @MattGal can comment on the situation.

Yes, it would seem you need a newer SDK to move past that.  The only other workaround would be to set `DOTNET_CLI_DO_NOT_USE_MSBUILD_SERVER=true` in the environment everywhere this build might need to work, which I think is undesirable.

dougbu committed:
Bump SDK version yet again

dougbu commented:
@JamesNK please see @vitek-karas's comments at https://github.com/dotnet/aspnetcore/pull/43028#issuecomment-1221610190 and later. I suspect you know how best to do something like dotnet/runtime#73238 in this repo. In any case, you're on build ops tomorrow&hellip;

JamesNK commented:
It shouldn't be required because [I already added a suppression for the warning](https://github.com/dotnet/aspnetcore/pull/43028/commits/6ad19f48b530a0dbe9a3a7915d12e371a40aa659).
```xml
    <!-- don't warn about unnecessary trim warning suppressions. can be removed with preview 6. -->
    <NoWarn>$(NoWarn);IL2121</NoWarn>
```
The NoWarn fixed all the failing builds except for `aspnetcore-ci (Build Source-Build (Managed))`. The error message from it doesn't give an indication about why it is failing.

Removing unnecessary suppressions is cleanup for the future. There are dozens. It will touch a lot of code and will take a bit of time.

dougbu commented:
Since aspnetcore-components-e2e and the source-build job are still failing, we need something in this branch.

In any case, @JamesNK I thought the recent comments / issues were about _redundant_ warning suppressions❔

JamesNK commented:
Yes, there are dozens of redundant warning suppressions. They showed up in builds before I added the NoWarn.

aspnetcore-components-e2e is failing because the SDK download is getting a 404 - https://dev.azure.com/dnceng/public/_build/results?buildId=1960417&view=logs&j=fe94c0c9-bb8c-5d6f-3b51-887173cc2f5c&t=7877c2d9-a5b1-59a3-dc2b-0d7e9d9db055

dougbu commented:
@JamesNK we've seen problems w/ Arcade retries lately. The initial 404s for the SDK itself should be ignored (looks like it was) and the actual failure seems later (when Arcade downloads runtime versions). I [asked about this](https://teams.microsoft.com/l/message/19:afba3d1545dd45d7b79f34c1821f6055@thread.skype/1661219678760?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=4d73664c-9f2f-450d-82a5-c2f02756606d&parentMessageId=1661219678760&teamName=.NET%20Core%20Eng%20Services%20Partners&channelName=First%20Responders&createdTime=1661219678760&allowXTenantAccess=false) in the First Responders channel.

/cc @dotnet/dnceng in case you want to respond here instead of FR

dkurepa commented:
As Matt mentioned above, it looks like it's a bug with the SDK, updating to a newer version should fix the `dotnet-install` issues. (based on the discussions in https://github.com/dotnet/arcade/issues/10311)

JamesNK commented:
@dkurepa This PR is currently updating the SDK to 7.0.100-rc.2.22419.24. CI builds continue to show download errors when trying to acquire the SDK:

```
Attempting to install dotnet from public_location.
dotnet-install: Note that the intended use of this script is for Continuous Integration (CI) scenarios, where:
dotnet-install: - The SDK needs to be installed without user interaction and without admin rights.
dotnet-install: - The SDK installation doesn't need to persist across multiple CI runs.
dotnet-install: To set up a development environment or to run apps, use installers rather than this script. Visit https://dotnet.microsoft.com/download to get the installer.

dotnet-install: Attempting to download using primary link https://dotnetcli.azureedge.net/dotnet/Sdk/7.0.100-rc.2.22419.24/dotnet-sdk-7.0.100-rc.2.22419.24-linux-x64.tar.gz
curl: (22) The requested URL returned error: 404 
dotnet-install: The resource at primary link 'https://dotnetcli.azureedge.net/dotnet/Sdk/7.0.100-rc.2.22419.24/dotnet-sdk-7.0.100-rc.2.22419.24-linux-x64.tar.gz' is not available.
dotnet-install: Attempting to download using legacy link https://dotnetcli.azureedge.net/dotnet/Sdk/7.0.100-rc.2.22419.24/dotnet-dev-ubuntu.18.04-x64.7.0.100-rc.2.22419.24.tar.gz
curl: (22) The requested URL returned error: 404 
dotnet-install: The resource at legacy link 'https://dotnetcli.azureedge.net/dotnet/Sdk/7.0.100-rc.2.22419.24/dotnet-dev-ubuntu.18.04-x64.7.0.100-rc.2.22419.24.tar.gz' is not available.
dotnet-install: Attempting to download using primary link https://dotnetbuilds.azureedge.net/public/Sdk/7.0.100-rc.2.22419.24/dotnet-sdk-7.0.100-rc.2.22419.24-linux-x64.tar.gz
dotnet-install: Extracting zip from https://dotnetbuilds.azureedge.net/public/Sdk/7.0.100-rc.2.22419.24/dotnet-sdk-7.0.100-rc.2.22419.24-linux-x64.tar.gz
dotnet-install: Installed version is 7.0.100-rc.2.22419.24
dotnet-install: Adding to current process PATH: `/home/vsts/work/1/s/.dotnet`. Note: This change will be visible only when sourcing script.
dotnet-install: Note that the script does not resolve dependencies during installation.
dotnet-install: To check the list of dependencies, go to https://docs.microsoft.com/dotnet/core/install, select your operating system and check the "Dependencies" section.
dotnet-install: Installation finished successfully.
  Determining projects to restore...
  Restored /home/vsts/.nuget/packages/microsoft.dotnet.arcade.sdk/7.0.0-beta.22419.1/tools/Tools.proj (in 5.4 sec).
  Determining projects to restore...
  Restored /home/vsts/work/1/s/eng/tools/GenerateFiles/GenerateFiles.csproj (in 1.4 sec).
  GenerateFiles -> /home/vsts/work/1/s/artifacts/bin/GenerateFiles/Directory.Build.props
  GenerateFiles -> /home/vsts/work/1/s/artifacts/bin/GenerateFiles/Directory.Build.targets
  GenerateFiles -> /home/vsts/work/1/s/.config/dotnet-tools.json
/home/vsts/.nuget/packages/microsoft.dotnet.arcade.sdk/7.0.0-beta.22419.1/tools/InstallDotNetCore.targets(15,5): error : dotnet-install failed [/home/vsts/.nuget/packages/microsoft.dotnet.arcade.sdk/7.0.0-beta.22419.1/tools/Tools.proj]
Build failed with exit code 1. Check errors above.
```

What SDK version is this fixed in?

dougbu commented:
Workaround may be to add `DOTNET_CLI_DO_NOT_USE_MSBUILD_SERVER: true` just below `COMPlus_DbgMiniDumpName: ...` in three places, starting at https://github.com/dotnet/aspnetcore/blob/f1353bd0de37a817f9f6acf22451ccd2d5c6a462/.azure/pipelines/jobs/default-build.yml#L226. This was mentioned recently but I didn't get to it.

dougbu committed:
Hack to stop using `msbuild` server

- please open an issue to keep trying to remove this

dougbu commented:
Added the hack I mentioned in latest commit.

@JamesNK please file an issue to remove these `!temporary! lines when we can.

/fyi @rainersigwald @rokonec you'll need @MattGal's simple repro instead of building this entire repo to see the runtime download issue 😉

MattGal commented:
> /fyi @rainersigwald @rokonec you'll need @MattGal's simple repro instead of building this entire repo to see the runtime download issue 😉

It actually only takes about 10 minutes to do the "real" scenario repro too with docker setup.  I'll make sure @rokonec has a usable one, one version or another.

JamesNK commented:
The current state is all checks are passing except `(Build Source-Build (Managed))`. Who is handling fixing it?

MattGal commented:
> The current state is all checks are passing except `(Build Source-Build (Managed))`. Who is handling fixing it?

This is an actual bug in the copy of MSBuild included with the SDK update, so it's probably still not a desirable SDK update to merge to your repo.  As of yesterday I think @rokonec is working on the fix.

dougbu commented:
> This is an actual bug in the copy of MSBuild included with the SDK update, so it's probably still not a desirable SDK update to merge to your repo

Not sure I agree because we're talking about the 'main' branch and some of our team are working on 8.0.0 features that need a newer SDK. Would rather expand the `DOTNET_CLI_DO_NOT_USE_MSBUILD_SERVER: true` thing short term. Could you try that @captainsafia❔

Basically, my https://github.com/dotnet/aspnetcore/pull/43028/commits/ac92df5a430f654555e499aedbbb8562e5cc1538 commit didn't affect source-build but should help everywhere else. Need a similar change in the source-build job in ci.yml.

Commit:
Disable msbuild server for source-build job

MattGal commented:
@dougbu there's a workaround in Arcade here: https://github.com/dotnet/arcade/pull/10585, that should allow you to use any of the SDKs with the breaking change without setting env vars.

captainsafia commented:
> @dougbu there's a workaround in Arcade here: https://github.com/dotnet/arcade/pull/10585, that should allow you to use any of the SDKs with the breaking change without setting env vars.

I noticed that you posted this comment (https://github.com/dotnet/arcade/pull/10585#issuecomment-1227742528) on the PR. I assume you'll be doing the validation for the fix on this branch?

dougbu commented:
possible (I am not looking at the moment) the Arcade update in release/7.0 hasn't flown into main yet. also possible it didn't include your latest fix @MattGal

MattGal commented:
> possible (I am not looking at the moment) the Arcade update in release/7.0 hasn't flown into main yet. also possible it didn't include your latest fix @MattGal

Nothing is merged into Arcade's main branch yet.  Locally https://github.com/dotnet/arcade/pull/10585 looks like the right fix, but I'm working through the other breaks in this PR now. What is clear is that it gets a little past the previous repro then is using some not-the-root nuget.config file that doesn't have the general-testing feed.  I'm trying sticking this feed into every nuget.config I find now.

MattGal commented:
I think the latest commit in https://github.com/dotnet/arcade/pull/10585  does the trick.  I'm trying out just the SDK update from this PR and that arcade update in this draft PR: https://github.com/dotnet/aspnetcore/pull/43568

captainsafia commented:
> I think the latest commit in [dotnet/arcade#10585](https://github.com/dotnet/arcade/pull/10585) does the trick. I'm trying out just the SDK update from this PR and that arcade update in this draft PR: #43568

@MattGal Any success with the trial? I saw the draft PR was closed but wasn't sure of the outcome.

MattGal commented:
> > I think the latest commit in [dotnet/arcade#10585](https://github.com/dotnet/arcade/pull/10585) does the trick. I'm trying out just the SDK update from this PR and that arcade update in this draft PR: #43568
> 
> @MattGal Any success with the trial? I saw the draft PR was closed but wasn't sure of the outcome.

Yes and no.  It definitely seemed to fix the problem you were hitting so I merged it to Arcade main.  Unfortunately there are access denied errors happening now in arcade-validation tests.  I'm investigating right now .

MattGal commented:
@captainsafia unfortunately this change breaks arcade-validation pretty clearly by itself.  The scenario might be a little weird (arcade-validation creates and deletes repos for testing, the deletion is where it got stuck) but it's clear we don't want to broadly roll out the workaround for now.  I think disabling the msbuild server via environment variable is unfortunately ASP.NET's best way forward right now.

dougbu commented:
Looks like this PR needs a few recent changes from main&hellip;

dougbu commented:
/azp run

dougbu commented:
> please file an issue to remove these `!temporary!` lines when we can.

@JamesNK and @captainsafia did either of you do this❔ I didn't see anything filed over the last few days but may not have gone back far enough.

captainsafia commented:
> @JamesNK and @captainsafia did either of you do this❔ I didn't see anything filed over the last few days but may not have gone back far enough.

I didn't create one because I wasn't sure how the explorations with the alternative fix would pan out. Given we're sticking with the workaround, it makes sense to file the issue.

BTW, [Disable msbuild server for source-build job](https://github.com/dotnet/aspnetcore/pull/43028/commits/77b2e43b5f9cf84479da4c1ce522adb1aa3d40b0) doesn't seem to have fixed the source build break. :/ Not sure if I placed the configuration incorrectly.

dougbu commented:
`DOTNET_CLI_DO_NOT_USE_MSBUILD_SERVER` environment variable is set correctly AFAICT but source-build job is still failing in the exact same way it has been. Is there something odd about the specific build container, not using job.yml, the script _not_ silently continuing after `curl: (22) The requested URL returned error: 404 Not Found`, &hellip;❔

I'm looking at [build #20220826.30](https://dev.azure.com/dnceng/public/_build/results?buildId=1970755&view=logs&jobId=2f0d093c-1064-5c86-fc5b-b7b1eca8e66a&j=2f0d093c-1064-5c86-fc5b-b7b1eca8e66a&t=d2b639a6-cb19-5f1f-66fd-8047f66b3026) which is currently on its way to fail due to this continuing problem. Error is shown in [this Build.binlog](https://artprodcus3.artifacts.visualstudio.com/Ab55de4ed-4b5a-4215-a8e4-0a0a5f71e7d8/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2RuY2VuZy9wcm9qZWN0SWQvOWVlNmQ0NzgtZDI4OC00N2Y3LWFhY2MtZjZlNmQwODJhZTZkL2J1aWxkSWQvMTk3MDc1NS9hcnRpZmFjdE5hbWUvQnVpbGRMb2dzX1NvdXJjZUJ1aWxkX01hbmFnZWRfQXR0ZW1wdDE1/content?format=file&subPath=%2Fartifacts%2Flog%2FRelease%2FBuild.binlog) as well.

Any suggestions much appreciated @MichaelSimons @rainersigwald @MattGal 

----

At the moment, I don't have a good guess because `DOTNET_CLI_DO_NOT_USE_MSBUILD_SERVER` isn't mentioned in any .binlog from the build that I checked. It feels like environment variables that aren't directly used in the build (e.g. `COMPlus_DbgEnableMiniDump` in our other jobs) aren't logged anymore. That leaves me w/ insufficient information to debug the problem.

The suggestion in Structured Log viewer is to
``` text
Define a value for MSBUILDLOGALLENVIRONMENTVARIABLES to log all environment variables. Only those used in evaluating properties are currently logged.
```
but that's very time-consuming when it takes a CI build to reproduce the problem. This feels like a regression @rainersigwald.

Should we unconditionally add `MSBUILDLOGALLENVIRONMENTVARIABLES: true` in all CI jobs that may produce binary logs❔

JamesNK commented:
> > please file an issue to remove these `!temporary!` lines when we can.
> 
> @JamesNK and @captainsafia did either of you do this❔ I didn't see anything filed over the last few days but may not have gone back far enough.

Done: https://github.com/dotnet/runtime/issues/74689

dougbu commented on .azure/pipelines/ci.yml:
```diff
@@ -836,6 +836,8 @@ stages:
             env:
               BLOB_CONTAINER_URL: $(source-dot-net-stage1-blob-container-url)
+              # !temporary! Remove as soon as .NET SDK includes a new-enough `msbuild` to make this hack unnecessary
+              DOTNET_CLI_DO_NOT_USE_MSBUILD_SERVER: true
```
```suggestion
              DOTNET_CLI_DO_NOT_USE_MSBUILD_SERVER: true
              # Log environment variables in binary logs to ease debugging
              MSBUILDLOGALLENVIRONMENTVARIABLES: true
```

dougbu committed:
Add `MSBUILDLOGALLENVIRONMENTVARIABLES`

- just in source-build job since that's the one that's failing
  - though, this might be useful everywhere

rainersigwald commented:
> At the moment, I don't have a good guess because `DOTNET_CLI_DO_NOT_USE_MSBUILD_SERVER` isn't mentioned in any .binlog from the build that I checked. It feels like environment variables that aren't directly used in the build (e.g. `COMPlus_DbgEnableMiniDump` in our other jobs) aren't logged anymore. That leaves me w/ insufficient information to debug the problem.
> 
> The suggestion in Structured Log viewer is to
> 
> ```
> Define a value for MSBUILDLOGALLENVIRONMENTVARIABLES to log all environment variables. Only those used in evaluating properties are currently logged.
> ```
> 
> but that's very time-consuming when it takes a CI build to reproduce the problem. This feels like a regression @rainersigwald.
> 
> Should we unconditionally add `MSBUILDLOGALLENVIRONMENTVARIABLES: true` in all CI jobs that may produce binary logs❔

cc @Forgind and @baronfel on this feedback--I think we should add an allowlist for `MSBUILD*` and `COMPlus_*` at the bare minimum.

dougbu commented:
It looks like source-build issues aren't the only problem. Not sure the exact problem but at least one `WebApiTemplateCSharp_IdentityWeb_SingleOrg_ProgramMain_BuildsAndPublishes` run failed in the last two builds. For example, in the [last build](https://dev.azure.com/dnceng/public/_build/results?buildId=1974093&view=results):

<details>
<summary>WebApiTemplateCSharp_IdentityWeb_IndividualB2C_ProgramMain_NoHttps_BuildsAndPublishes(auth: "IndividualB2C", args: ["--use-program-main", "--no-https"]) on Ubuntu 18.04</summary>

``` text
Project new webapi  --auth IndividualB2C --use-program-main --no-https failed to build. Exit code 139.
/datadisks/disk1/work/AA3D0981/p/dotnet-cli/dotnet build --no-restore -c Debug /bl
StdErr:
StdOut: MSBuild version 17.4.0-preview-22413-01+f0a66ec43 for .NET
/datadisks/disk1/work/AA3D0981/p/dotnet-cli/sdk/7.0.100-rc.2.22419.24/MSBuild.dll -consoleloggerparameters:Summary -maxcpucount -property:Configuration=Debug -verbosity:m /bl ./AspNet.uf1imc03i0z4.csproj
/datadisks/disk1/work/AA3D0981/p/dotnet-cli/sdk/7.0.100-rc.2.22419.24/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(219,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/datadisks/disk1/work/AA3D0981/w/A8AC0938/e/Templates/BaseFolder/AspNet.uf1imc03i0z4/AspNet.uf1imc03i0z4.csproj]
AspNet.uf1imc03i0z4 -> /datadisks/disk1/work/AA3D0981/w/A8AC0938/e/Templates/BaseFolder/AspNet.uf1imc03i0z4/bin/Debug/net7.0/AspNet.uf1imc03i0z4.dll

Build succeeded.
0 Warning(s)
0 Error(s)

Time Elapsed 00:00:00.55

Expected: True
Actual:   False

Stack trace
   at Templates.Mvc.Test.WebApiTemplateTest.PublishAndBuildWebApiTemplate(String languageOverride, String auth, String[] args) in /_/src/ProjectTemplates/test/Templates.Mvc.Tests/WebApiTemplateTest.cs:line 254
--- End of stack trace from previous location ---
```

Core dump at <https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-aspnetcore-refs-pull-43028-merge-9a8cf838a9654554a1/Templates.Mvc.Tests--net7.0/1/core.1000.19678?helixlogtype=result>.
</details>

<details>
<summary>WebApiTemplateCSharp_IdentityWeb_SingleOrg_ProgramMain_BuildsAndPublishes(auth: "SingleOrg", args: ["--use-program-main"]) on OSX 11</summary>

``` text
Project new webapi  --auth SingleOrg --use-program-main failed to create/restore. Exit code -1.
/private/tmp/helix/working/B2D209D2/p/dotnet-cli/dotnet new webapi  --debug:disable-sdk-templates --debug:custom-hive "/private/tmp/helix/working/B2D209D2/w/AF87098C/e/Hives/.templateEngine" --auth SingleOrg --use-program-main -o /tmp/helix/working/B2D209D2/w/AF87098C/e/Templates/BaseFolder/AspNet.Watjhorfp2bd
StdErr: Restore failed.
Post action failed.
Manual instructions: Run 'dotnet restore'

StdOut: The template "ASP.NET Core Web API" was created successfully.

Processing post-creation actions...
Restoring /tmp/helix/working/B2D209D2/w/AF87098C/e/Templates/BaseFolder/AspNet.Watjhorfp2bd/AspNet.Watjhorfp2bd.csproj:
Determining projects to restore...
Restored /tmp/helix/working/B2D209D2/w/AF87098C/e/Templates/BaseFolder/AspNet.Watjhorfp2bd/AspNet.Watjhorfp2bd.csproj (in 46 ms).

Expected: True
Actual:   False


Stack trace
   at Templates.Mvc.Test.WebApiTemplateTest.PublishAndBuildWebApiTemplate(String languageOverride, String auth, String[] args) in /_/src/ProjectTemplates/test/Templates.Mvc.Tests/WebApiTemplateTest.cs:line 254
--- End of stack trace from previous location ---
```
</details>

Who has a good understanding of these issues or the specific tests❔

dougbu commented:
On the continuing source-build problem, I suspect our environment settings are not working at all. See [Build.repotasks.binlog](https://artprodcus3.artifacts.visualstudio.com/Ab55de4ed-4b5a-4215-a8e4-0a0a5f71e7d8/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_apis/artifact/cGlwZWxpbmVhcnRpZmFjdDovL2RuY2VuZy9wcm9qZWN0SWQvOWVlNmQ0NzgtZDI4OC00N2Y3LWFhY2MtZjZlNmQwODJhZTZkL2J1aWxkSWQvMTk3NDA5My9hcnRpZmFjdE5hbWUvQnVpbGRMb2dzX1NvdXJjZUJ1aWxkX01hbmFnZWRfQXR0ZW1wdDE1/content?format=file&subPath=%2Fartifacts%2Fsource-build%2Fself%2Fsrc%2Fartifacts%2Flog%2FDebug%2FBuild.repotasks.binlog). That and other binary logs are not picking up `MSBUILDLOGALLENVIRONMENTVARIABLES` set in the build step.

dougbu committed:
Correct source-build job

- move new `env:` settings out of `SourceIndexUpload` job
- use `variables:` instead because jobs don't support `env:`

dougbu commented:
> I suspect our environment settings are not working at all

Found it❕

captainsafia commented:
> Who has a good understanding of these issues or the specific tests❔

The log seems to indicate that there might have been a segfault when running `dotnet new`. Is this happening persistently? If so, might indicate an issue with the version of the SDK we pick up in this PR.

> Project new webapi  --auth IndividualB2C --use-program-main --no-https failed to build. Exit code 139.

wtgodbe committed:
Update global.json

dougbu commented:
@captainsafia you're right and I think the latest build hit an outage or two.
@wtgodbe please do not update the SDK version again. It's doubtful we'll get a fix for the source-build problem from there. Instead, we need to figure out why YAML variables aren't becoming environment variables and affecting source-build. I'm looking&hellip;

dougbu committed:
!fixup! Correct placement of source-build job properties

dougbu commented:
Build w/ @wtgodbe's latest SDK update had already failed w/ usual source-build problem when I pushed my newest fixup. Earlier placement of the job variables and even-earlier attempt to control the source-build job timeout wasn't quite right.

wtgodbe commented:
`Templates.Blazor.Test.BlazorWasmTemplateAuthTest.BlazorWasmHostedTemplate_IndividualAuth_NoHttps_Works_WithOutLocalDB` failed, looks like it may be related to the SDK:

> Project  failed to publish. Exit code 1.
/private/tmp/helix/working/BD1F09DF/p/dotnet-cli/dotnet publish --no-restore -c Release /bl
StdErr:
StdOut: MSBuild version 17.4.0-preview-22416-02+5d102ae37 for .NET
/private/tmp/helix/working/BD1F09DF/p/dotnet-cli/sdk/7.0.100-rc.2.22426.5/MSBuild.dll -maxcpucount -property:_IsPublishing=true -property:Configuration=Release -target:Publish -verbosity:m /bl ./AspNet.dvhepurhcgwo.Server.csproj
/private/tmp/helix/working/BD1F09DF/p/dotnet-cli/sdk/7.0.100-rc.2.22426.5/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(219,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/private/tmp/helix/working/BD1F09DF/w/A2D0090F/e/Templates/BaseFolder/AspNet.dvhepurhcgwo/Server/AspNet.dvhepurhcgwo.Server.csproj]
/private/tmp/helix/working/BD1F09DF/p/dotnet-cli/sdk/7.0.100-rc.2.22426.5/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.RuntimeIdentifierInference.targets(219,5): message NETSDK1057: You are using a preview version of .NET. See: https://aka.ms/dotnet-support-policy [/private/tmp/helix/working/BD1F09DF/w/A2D0090F/e/Templates/BaseFolder/AspNet.dvhepurhcgwo/Shared/AspNet.dvhepurhcgwo.Shared.csproj]
AspNet.dvhepurhcgwo.Shared -> /private/tmp/helix/working/BD1F09DF/w/A2D0090F/e/Templates/BaseFolder/AspNet.dvhepurhcgwo/Shared/bin/Release/net7.0/AspNet.dvhepurhcgwo.Shared.dll
AspNet.dvhepurhcgwo.Client -> /private/tmp/helix/working/BD1F09DF/w/A2D0090F/e/Templates/BaseFolder/AspNet.dvhepurhcgwo/Client/bin/Release/net7.0/AspNet.dvhepurhcgwo.Client.dll
AspNet.dvhepurhcgwo.Client (Blazor output) -> /private/tmp/helix/working/BD1F09DF/w/A2D0090F/e/Templates/BaseFolder/AspNet.dvhepurhcgwo/Client/bin/Release/net7.0/wwwroot
AspNet.dvhepurhcgwo.Server -> /private/tmp/helix/working/BD1F09DF/w/A2D0090F/e/Templates/BaseFolder/AspNet.dvhepurhcgwo/Server/bin/Release/net7.0/AspNet.dvhepurhcgwo.Server.dll
Optimizing assemblies for size may change the behavior of the app. Be sure to test after publishing. See: https://aka.ms/dotnet-illink
Compressing Blazor WebAssembly publish artifacts. This may take a while...
AspNet.dvhepurhcgwo.Server -> /private/tmp/helix/working/BD1F09DF/w/A2D0090F/e/Templates/BaseFolder/AspNet.dvhepurhcgwo/Server/bin/Release/net7.0/publish/

> Expected: True
Actual:   False

@javiercn @TanayParikh @MackinnonBuck any ideas? rerunning for now.

dougbu commented:
As @captainsafia [commented earlier](https://github.com/dotnet/aspnetcore/pull/43028#issuecomment-1231914642) in response to my "Who has a good understanding of these issues or the specific tests❔" [comment](https://github.com/dotnet/aspnetcore/pull/43028#issuecomment-1230805316), it appears our old `dotnet` seg fault problems have returned. But the scenarios might be narrower this time around. What is consistent between the latest failure and the ones I mentioned❔

@dsplaisted @marcpopMSFT who are the best contacts for `dotnet` seg faults❔ If our tests are doing something unexpected, please let us know.

dougbu commented:
Note we need to track fixes for the "exit code 139" and `msbuild` problems. We have a new SDK but builds of main will likely be less reliable until those problems are corrected.

dougbu commented:
Put another way, #43581 tracks one problem we're going to see sporadically in main until something makes updates `msbuild` in the .NET SDK.

Separately, dotnet/runtime#68393 (or some of the issues it references) has regressed again or we've hit a new variant of the problem. @jkotas @rainersigwald @radical @lewing @bartonjs @vcsjones @karelz you were all involved in some of those `dotnet` seg fault / exit code 139 issues. Details mostly in the first attempt of the CI for this PR and comments above e.g.

- https://github.com/dotnet/aspnetcore/pull/43028#issuecomment-1230805316
- https://github.com/dotnet/aspnetcore/pull/43028#issuecomment-1231914642
- https://github.com/dotnet/aspnetcore/pull/43028#issuecomment-1232084050

We may need a new issue to track addressing this.

Thoughts❔

jkotas commented:
> dotnet seg fault / exit code 139 issues

This is common failure mode with many different root causes. We need either crash dump or repro steps to make progress on it.

dougbu commented:
@jkotas there's a dump at https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-aspnetcore-refs-pull-43028-merge-9a8cf838a9654554a1/Templates.Mvc.Tests--net7.0/1/core.1000.19678?helixlogtype=result and likely one in the artifacts for the test(s) that failed during the first attempt at the last build of this PR.

jkotas commented:
> https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-aspnetcore-refs-pull-43028-merge-9a8cf838a9654554a1/Templates.Mvc.Tests--net7.0/1/core.1000.19678?helixlogtype=result 

The stacktrace of the crash is:
```
libcoreclr!sigsegv_handler+0x260 [/__w/1/s/src/coreclr/pal/src/exception/signal.cpp @ 15732480] 
libpthread_2_27!_restore_rt
libpthread_2_27!_GI___pthread_rwlock_wrlock+0x12
libcrypto_so_1!CRYPTO_THREAD_write_lock+0x9
libcrypto_so_1!RAND_get_rand_method+0x33
libcrypto_so_1!RAND_bytes+0x10
libcrypto_so_1!EVP_MD_CTX_ctrl+0x132f
libcrypto_so_1!EVP_CIPHER_CTX_ctrl+0x17
libssl_so_1!SSL_in_before+0x13bd4
libssl_so_1!SSL_in_before+0x59ea
libssl_so_1!SSL_in_before+0x3d6
libssl_so_1!SSL_do_handshake+0x54
libSystem_Security_Cryptography_Native_OpenSsl!CryptoNative_SslDoHandshake+0x25 [/__w/1/s/src/native/libs/System.Security.Cryptography.Native/pal_ssl.c @ 555] 
```
This is dup of https://github.com/dotnet/runtime/issues/68393. It is bug in Ubuntu https://bugs.launchpad.net/ubuntu/+source/openssl/+bug/1983100 that is in the process of being fixed.

dougbu commented:
Will the problem automatically go away @jkotas if we migrate away from Ubuntu 18.04 build / test agents❔ I doubt that's viable in general (yet) but we at least plan to stop using that version for our `container` builds and where we use AzDO-hosted build agents.

bartonjs commented:
The only persistent SIGSEGV that I personally know about is unique to Ubuntu 18.04, and is caused by tools having background TLS connections still running when the process exits.  Ubuntu has a bug on it (https://bugs.launchpad.net/ubuntu/+source/openssl/+bug/1983100), but it sadly hasn't hit an acceptable final release yet.

So... yes, moving off of Ubuntu 18.04 (or their fix finally rolling out) would probably take the segfaults away.

dougbu commented:
In a separate bit of follow-up, I'm looking at including `MSBUILDLOGALLENVIRONMENTVARIABLES` in almost every job we submit. Trying to decide whether to target release/7.0 or main w/ that change&hellip;

@rainersigwald @Forgind and @baronfel, environment variables are still logged in full in recent release/7.0 builds. Is that likely to change as new `msbuild` bits get into the RC2 or RTM 7.0.0 SDKs or newer Visual Studio releases❔ If yes, I'll target release/7.0 proactively; otherwise, just add the workaround in main.

Forgind commented:
Yes; the change should flow with the new MSBuild, so if it's causing trouble for you, you should target release/7.0 with that.

We're also discussing the best way to communicate clearly if we make certain environment variables always logged. If you're interested, I'd be happy to add you to that conversation.

dougbu commented:
> I'd be happy to add you to that conversation.

I probably don't have much more to add beyond my "regression" thoughts above but sure&hellip;

dougbu commented:
#43644 opened, targeting release/7.0. I'll flow automatically (modulo an expected merge conflict) into main.
